tests/datalab/test_datalab.py

```diff
@@ -932,3 +932,41 @@ def test_find_issues_features_works_with_and_without_labels(self, features, labe
 
         pd.testing.assert_frame_equal(issues_without_labels, issues_with_labels)
         pd.testing.assert_frame_equal(issues_without_labels, issues_without_label_name)
+
+
+class TestDataLabClassImbalanceIssues:
+    K = 3
+    N = 100
+    num_features = 2
+
+    @pytest.fixture
+    def random_embeddings(self):
+        np.random.seed(SEED)
+        return np.random.rand(self.N, self.num_features)
+
+    @pytest.fixture
+    def imbalance_labels(self):
+        np.random.seed(SEED)
+        labels = np.random.choice(np.arange(self.K - 1), 100, p=[0.5] * (self.K - 1))
+        labels[0] = 2
+        return labels
+
+    @pytest.fixture
+    def pred_probs(self):
+        np.random.seed(SEED)
+        pred_probs_array = np.random.rand(self.N, self.K)
+        return pred_probs_array / pred_probs_array.sum(axis=1, keepdims=True)
+
+    def test_incremental_search(self, pred_probs, random_embeddings, imbalance_labels):
+        data = {"labels": imbalance_labels}
+        lab = Datalab(data=data, label_name="labels")
+        lab.find_issues(pred_probs=pred_probs, issue_types={"label": {}})
+        summary = lab.get_issue_summary()
+        assert len(summary) == 1
+        assert "class_imbalance" not in summary["issue_type"].values
+        lab.find_issues(features=random_embeddings, issue_types={"class_imbalance": {}})
+        summary = lab.get_issue_summary()
+        assert len(summary) == 2
+        assert "class_imbalance" in summary["issue_type"].values
+        class_imbalance_summary = lab.get_issue_summary("class_imbalance")
+        assert class_imbalance_summary["num_issues"].values[0] > 0
```

Comment:
This is a very comprehensive test.

It seems to me that `ClassImbalanceIssues.find_issues` doesn't require any arguments to be run (it just needs the labels provided in the associated `Datalab` instance.

Can you make a similar test that just runs

```python
 lab = Datalab(data=data, label_name="labels")
lab.find_issues( issue_types={"class_imbalance": {}})
```

I expect the resulting class imbalance issues to be the same as in this test.

Reply:
Got it. Is this part of the incremental search or a separate test?

Reply:
A separate test. Some code repetition is fine here.

Reply:
Related question: For `ClassImbalanceIssues.find_issues` tests (i.e. test_imbalance.py), I do pass the features to the function. Should I remove them from those tests, and subsequently remove `kwargs` from the `find_issues` function of `ClassImbalanceIssueManager`? Since I can add `"class_imbalance":{}`  here - 

https://github.com/cleanlab/cleanlab/blob/1575ba743231dc741076cfc6c37f195f27e80442/cleanlab/datalab/issue_finder.py#L217C1-L221

Reply:
@tataganesh Good question!

Removing `features` from the issue manager tests would be good, but I believe keeping `kwargs` is safer for now.

I don't think it's necessary to update the `args_dict` in the linked code.